File: Linked List /C++/8. 234. Palindrome Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        vector<int> stack;
        ListNode* temp = head; 
        while(temp != NULL)
        {
            stack.push_back(temp -> val);
            temp = temp->next;
        }
        temp = head;
        while(temp != NULL)
        {
            if(temp -> val != stack.back())
            {
                return false;
            }
            stack.pop_back();
            temp = temp -> next;
        }
        return true;
    }
};

File: Linked List /Python/8. 234. Palindrome Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        temp = head
        stack = []
        isPalin = True
        while temp is not None:
            stack.append(temp)
        
        temp = head
        while temp is not None:
            if temp.val != stack.end():
                isPalin == False
            stack.pop()
            temp = temp.next
            isPalin = True
        return isPalin        
